File: ios/IPCClient/IPCClient/ViewController.m
//
//  ViewController.m
//  IPCClient
//
//  Created by 村上 幸雄 on 12/07/29.
//  Copyright (c) 2012年 Bitz Co., Ltd. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

@synthesize textField = _textField;
@synthesize button = _button;

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)viewDidUnload
{
    [super viewDidUnload];
    // Release any retained subviews of the main view.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return (interfaceOrientation != UIInterfaceOrientationPortraitUpsideDown);
}

- (IBAction)send:(id)sender
{
    NSURL   *url = [NSURL URLWithString:@"IPCServer.demo://IPCServer.demo?key=value"];
    if ([[UIApplication sharedApplication] canOpenURL:url]) {
        UIPasteboard    *pastedboard = [UIPasteboard pasteboardWithName:@"demo.IPCClient" create:YES];
        pastedboard.persistent = YES;
        [pastedboard setString:self.textField.text];
        [[UIApplication sharedApplication] openURL:url];
    }
}

@end

File: ios/Identifier/Identifier/ViewController.swift
//
//  ViewController.swift
//  Identifier
//
//  Created by 村上幸雄 on 2017/04/23.
//  Copyright © 2017年 Bitz Co., Ltd. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    @IBOutlet weak var label: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        self.label.text = ""
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func get(sender: AnyObject) {
        self.label.text = Identifier.sharedInstance.uuid
    }
    
    @IBAction func reset(sender: AnyObject) {
        Identifier.sharedInstance.reset()
    }
    
    @IBAction func update(sender: AnyObject) {
        let uuidString = NSUUID().uuidString
        Identifier.sharedInstance.update(uuidString: uuidString)
    }

}

